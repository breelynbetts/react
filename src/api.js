const apiHost = () => {} // No-op in our mock version.

const searchNasa = () => Promise.resolve({
  data: [{
      "id": 689597,
      "sol": 2540,
      "camera": {
        "id": 20,
        "name": "FHAZ",
        "rover_id": 5,
        "full_name": "Front Hazard Avoidance Camera"
      },
      "img_src": "https://mars.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/02540/opgs/edr/fcam/FLB_622990123EDR_F0763002FHAZ00341M_.JPG",
      "earth_date": "2019-09-28",
      "rover": {
        "id": 5,
        "name": "Curiosity",
        "landing_date": "2012-08-06",
        "launch_date": "2011-11-26",
        "status": "active",
        "max_sol": 2540,
        "max_date": "2019-09-28",
        "total_photos": 366206,
        "cameras": [{
            "name": "FHAZ",
            "full_name": "Front Hazard Avoidance Camera"
          },
          {
            "name": "NAVCAM",
            "full_name": "Navigation Camera"
          },
          {
            "name": "MAST",
            "full_name": "Mast Camera"
          },
          {
            "name": "CHEMCAM",
            "full_name": "Chemistry and Camera Complex"
          },
          {
            "name": "MAHLI",
            "full_name": "Mars Hand Lens Imager"
          },
          {
            "name": "MARDI",
            "full_name": "Mars Descent Imager"
          },
          {
            "name": "RHAZ",
            "full_name": "Rear Hazard Avoidance Camera"
          }
        ]
      }
    },
    {
      "id": 689598,
      "sol": 2540,
      "camera": {
        "id": 20,
        "name": "FHAZ",
        "rover_id": 5,
        "full_name": "Front Hazard Avoidance Camera"
      },
      "img_src": "https://mars.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/02540/opgs/edr/fcam/FRB_622990123EDR_F0763002FHAZ00341M_.JPG",
      "earth_date": "2019-09-28",
      "rover": {
        "id": 5,
        "name": "Curiosity",
        "landing_date": "2012-08-06",
        "launch_date": "2011-11-26",
        "status": "active",
        "max_sol": 2540,
        "max_date": "2019-09-28",
        "total_photos": 366206,
        "cameras": [{
            "name": "FHAZ",
            "full_name": "Front Hazard Avoidance Camera"
          },
          {
            "name": "NAVCAM",
            "full_name": "Navigation Camera"
          },
          {
            "name": "MAST",
            "full_name": "Mast Camera"
          },
          {
            "name": "CHEMCAM",
            "full_name": "Chemistry and Camera Complex"
          },
          {
            "name": "MAHLI",
            "full_name": "Mars Hand Lens Imager"
          },
          {
            "name": "MARDI",
            "full_name": "Mars Descent Imager"
          },
          {
            "name": "RHAZ",
            "full_name": "Rear Hazard Avoidance Camera"
          }
        ]
      }
    },
    {
      "id": 689599,
      "sol": 2540,
      "camera": {
        "id": 21,
        "name": "RHAZ",
        "rover_id": 5,
        "full_name": "Rear Hazard Avoidance Camera"
      },
      "img_src": "https://mars.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/02540/opgs/edr/rcam/RLB_622990164EDR_F0763002RHAZ00341M_.JPG",
      "earth_date": "2019-09-28",
      "rover": {
        "id": 5,
        "name": "Curiosity",
        "landing_date": "2012-08-06",
        "launch_date": "2011-11-26",
        "status": "active",
        "max_sol": 2540,
        "max_date": "2019-09-28",
        "total_photos": 366206,
        "cameras": [{
            "name": "FHAZ",
            "full_name": "Front Hazard Avoidance Camera"
          },
          {
            "name": "NAVCAM",
            "full_name": "Navigation Camera"
          },
          {
            "name": "MAST",
            "full_name": "Mast Camera"
          },
          {
            "name": "CHEMCAM",
            "full_name": "Chemistry and Camera Complex"
          },
          {
            "name": "MAHLI",
            "full_name": "Mars Hand Lens Imager"
          },
          {
            "name": "MARDI",
            "full_name": "Mars Descent Imager"
          },
          {
            "name": "RHAZ",
            "full_name": "Rear Hazard Avoidance Camera"
          }
        ]
      }
    }
  ]
})

const searchAstronomyPicture = () => Promise.resolve({
  data: [{
    "copyright": "Eric Wagner",
    "date": "2019-10-10",
    "explanation": "On September 24, a late evening commercial flight from Singapore to Australia offered stratospheric views of the southern hemisphere's night sky, if you chose a window seat. In fact, a well-planned seating choice with a window facing toward the Milky Way allowed the set up of a sensitive digital camera on a tripod mount to record the galaxy's central bulge in a series of 10 second long exposures.  By chance, one of the exposures caught this bright fireball meteor in the starry frame. Reflected along the wing of the A380 aircraft, the brilliant greenish streak is also internally reflected in the double layer window, producing a fainter parallel to the original meteor track. In the southern sky Jupiter is the bright source beneath the galactic bulge and seen next to a green beacon, just off the wing tip.",
    "hdurl": "https://apod.nasa.gov/apod/image/1910/MWBolideEricWagner2400.jpg",
    "media_type": "image",
    "service_version": "v1",
    "title": "Mid-Air Meteor and Milky Way",
    "url": "https://apod.nasa.gov/apod/image/1910/MWBolideEricWagner1200.jpg"
  }]

})

const searchEvents = () => Promise.resolve({
  data: [{
    "title": "EONET Events",
    "description": "Natural events from EONET.",
    "link": "https://eonet.sci.gsfc.nasa.gov/api/v2.1/events",
    "events": [
      {
        "id": "EONET_4475",
        "title": "Wildfires - Los Angeles County (Tick Fire), California, United States",
        "description": "",
        "link": "https://eonet.sci.gsfc.nasa.gov/api/v2.1/events/EONET_4475",
        "categories": [
          {
            "id": 8,
            "title": "Wildfires"
          }
        ],
        "sources": [
          {
            "id": "PDC",
            "url": "http://emops.pdc.org/emops/?hazard_id=96981"
          }
        ],
        "geometries": [
          {
            "date": "2019-10-24T18:16:00Z",
            "type": "Point",
            "coordinates": [
              -118.39468413,
              34.457634347
            ]
          }
        ]
      },
      {
        "id": "EONET_4476",
        "title": "Wildfires - San Bernardino County (Old Water Fire), California, United States",
        "description": "",
        "link": "https://eonet.sci.gsfc.nasa.gov/api/v2.1/events/EONET_4476",
        "categories": [
          {
            "id": 8,
            "title": "Wildfires"
          }
        ],
        "sources": [
          {
            "id": "PDC",
            "url": "http://emops.pdc.org/emops/?hazard_id=96977"
          }
        ],
        "geometries": [
          {
            "date": "2019-10-24T15:11:00Z",
            "type": "Point",
            "coordinates": [
              -117.26913,
              34.17922
            ]
          }
        ]
      },
      {
        "id": "EONET_4474",
        "title": "Wildfire -Sonoma County (Kincade Fire), California, United States",
        "description": "",
        "link": "https://eonet.sci.gsfc.nasa.gov/api/v2.1/events/EONET_4474",
        "categories": [
          {
            "id": 8,
            "title": "Wildfires"
          }
        ],
        "sources": [
          {
            "id": "PDC",
            "url": "http://emops.pdc.org/emops/?hazard_id=96971"
          }
        ],
        "geometries": [
          {
            "date": "2019-10-24T13:32:00Z",
            "type": "Point",
            "coordinates": [
              -122.807158563,
              38.758962281
            ]
          }
        ]
      }
    ]
  }  
  ]

})

export {
  apiHost,
  searchNasa,
  searchAstronomyPicture,
  searchEvents
}
