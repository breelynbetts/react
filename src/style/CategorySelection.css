@import url('https://fonts.googleapis.com/css?family=Montserrat|Staatliches&display=swap');

.eventName {
    font-family: 'Staatliches', cursive;
    font-size: 20px;
}

.eventDiv {
    width: 50%;
    margin-left: 25%;
    margin-bottom: 15px;
    border: 1px solid black;
}

.eventDiv:first-child {
    margin-top: 35px;
}

.eventData {
    border-bottom: 0.5px dashed black;
    width: 60%;
    margin-left: 20%;
}

.eventData:last-child {
    border-bottom: none;
}