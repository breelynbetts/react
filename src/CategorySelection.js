import React, { useEffect, useState } from 'react'
import { searchEvents } from './api.js'

import './style/CategorySelection.css'
import { apiHost } from './api'

import moment from 'moment'; 
// import {MapContainer, Map} from './components/Map.js'

function CategorySelection() {
    const [data, setData] = useState([])
    const [error, setError] = useState(null)
    const [locations, setLocations] = useState([])
    
    useEffect(() => apiHost('http://localhost:4000'))

    useEffect(() => {
        const fetchData = async () => {
            try {
                const result = await searchEvents({limit: 5});
                setData(result.events)
                setLocations(result.events)
            } catch (e) {
                setError('Sorry, but something went wrong.')
            }
        }
        fetchData();
      }, []);

    return (
        <div>
            {/* {locations === undefined ? <div>Working on displaying map ... </div> :
                <MapContainer>
                    <Map results={locations}/>
                </MapContainer>
            } */}
           <span >
           {data === undefined ? <div>Loading ...</div> : 
            <div>
                {data.map((event) => {
                    return (
                        <div key={event.id} className='eventDiv'>
                            <h4 className='eventName'key={event.id}>{event.title}</h4 >   
                            {event.geometries.map((loc) => {
                                return (
                                    <div className='eventData' key={loc.date}>
                                        <p>{moment(loc.date).format('LLLL')}</p>
                                        <p>location: 
                                           lat:  {loc.coordinates[0]}, lng:  { loc.coordinates[1] }
                                        </p>                            
                                    </div>
                                )
                            })}
                        </div>
                    )
                })}
            </div>
            }
            {error && (
                <div className="error">
                    {error}
                </div>
            )}
       </span>
    </div>
    )
}


export default CategorySelection