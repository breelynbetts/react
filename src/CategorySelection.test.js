import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

import sinon from 'sinon'

import * as api from './api'
import CategorySelection from './CategorySelection'


it('should start with an empty search field', () => {
  const component = TestRenderer.create(<CategorySelection />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

const setupAndLoadPage = async () => {
    const div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(<CategorySelection />, div)
    })
  
    // const searchInput = div.querySelector('input')
    // ReactTestUtils.act(() => {
    //   searchInput.value = 'hello'
    //   ReactTestUtils.Simulate.change(searchInput)
    // })
  
    // const searchForm = div.querySelector('form')
    // await ReactTestUtils.act(async () => {
    //   await ReactTestUtils.Simulate.submit(searchForm)
    // })
  
    return div
  }

describe('API calls', () => {
    let div
    beforeEach(async () => {
      sinon.stub(api, 'searchEvents')
  
      // To manage size, we supply a mock response that contains _only_ what the app will need. This does mean
      // that we need to revise the mock response if our app starts using more (or different) data.
      api.searchEvents.returns(Promise.resolve({
        data: [
          {
            id: 'FiGiRei2ICzzG',
            source_tld: 'tumblr.com',
            images: {
              fixed_width: {
                url: 'http://media2.giphy.com/media/FiGiRei2ICzzG/200w.gif'
              }
            }
          }
        ]
      }))
  
      div = await setupAndLoadPage()
    })
  
    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div)
      api.searchEvents.restore()
    })
  
  })
  
  describe('failed API calls', () => {
    let div
    beforeEach(async () => {
      sinon.stub(api, 'searchEvents')
      api.searchEvents.returns(Promise.reject('Mock failure'))
  
      div = await setupAndLoadPage()
    })
  
    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div)
      api.searchEvents.restore()
    })
  
    it('should display an alert when the API call fails', () => {
      // The document should contain the error div.
      const searchError = div.querySelector('div.error')
      expect(searchError.textContent).toEqual('Sorry, but something went wrong.')
    })
  })