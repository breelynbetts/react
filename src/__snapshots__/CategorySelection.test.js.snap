// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start with events on the page 1`] = `
<div>
  <span>
    <div />
  </span>
</div>
`;
